dali/test/dali_operator_test.h

```diff
@@ -0,0 +1,165 @@
+#ifndef DALI_DALI_OPERATOR_TEST_H
+#define DALI_DALI_OPERATOR_TEST_H
+
+#include <gtest/gtest.h>
+#include "dali/pipeline/pipeline.h"
+#include "dali/pipeline/data/backend.h"
+
+namespace dali {
+
+namespace util {
+
+template<typename T>
+struct is_tuple : std::false_type {
+};
+
+template<typename... T>
+struct is_tuple<std::tuple<T...>> : std::true_type {
+};
+
+
+/// This is a helper function, because there's no `if constexpr` in C++11
+template<typename Backend>
+std::string BackendStringName() {
+  DALI_FAIL("Selected Backend is not supported");
+}
+
+
+template<>
+std::string BackendStringName<CPUBackend>() {
+  return "cpu";
+}
+
+
+template<>
+std::string BackendStringName<GPUBackend>() {
+  return "gpu";
+}
+
+}  // namespace util
+
+template<typename Inputs, typename Outputs>
+class DaliOperatorTest : public ::testing::Test {
+//    static_assert(util::is_tuple<Inputs>::value, "Inputs has to be either a tuple or a void");
+//    static_assert(util::is_tuple<Outputs>::value, "Outputs has to be a tuple");
+
+public:
+  using Arguments =  std::map<std::string, float>; // TODO boost::any
```

Comment:
I see two other approaches that may be applied here:
One would be to specify Arguments as template parameters for each test - you won't have to resort to type erasure techniques.
The other one would be to somehow utilize information already specified in Schema, but this is rather vague suggestion.

Still, I would try to come up with something that wouldn't force the people that want to write an operator test, to basically copy the contents of schema multiple times to specify different Test.

Reply:
What do you mean by copying contents of schema? Could you specify?